spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/ScopePassingSpanSubscriber.java

```diff
@@ -46,65 +44,70 @@
 
 	private final Tracing tracing;
 	private final CurrentTraceContext currentTraceContext;
-	private final TraceContext root;
+	private final TraceContext traceContext;
 
 	private Subscription s;
 
 	ScopePassingSpanSubscriber(Subscriber<? super T> subscriber, Context ctx,
-			Tracing tracing) {
+	                           Tracing tracing) {
 		this.subscriber = subscriber;
 		this.tracing = tracing;
+		Tracer tracer = tracing.tracer();
 		this.currentTraceContext = this.tracing.currentTraceContext();
-		TraceContext root = ctx != null ? ctx.getOrDefault(TraceContext.class, this.currentTraceContext.get()) : null;
-		this.root = root;
-		this.context = ctx != null && root != null ? ctx.put(TraceContext.class, root)
+		Span rootSpan = ctx != null ? ctx.getOrDefault(Span.class, tracer.currentSpan())
```

Comment:
hmm why are we putting both in context? both Span.class and TraceContext.class? Is there other code using one or the other? If so, please make comment to what that code is.. probably we should only do one as it is double overhead to do both.

Reply:
ex `tracer.currentSpan()` inside calls `currentTraceContext.get()`